Library/Homebrew/utils/lock.sh

```diff
@@ -44,9 +44,10 @@ _create_lock() {
   [[ -x "${ruby}" ]] || ruby="$(type -P ruby)"
   [[ -x "${python}" ]] || python="$(type -P python)"
 
-  if [[ -x "${ruby}" ]] && "${ruby}" -e "exit(RUBY_VERSION >= '1.8.7')"
+  # Use /dev/stdin, otherwise Ruby can error if uid != euid.
+  if [[ -x "${ruby}" ]] && "${ruby}" /dev/stdin <<<"exit(RUBY_VERSION >= '1.8.7')"
```

Comment:
```suggestion
  if [[ -x "${ruby}" ]] && "${ruby}" - <<<"exit(RUBY_VERSION >= '1.8.7')"
```
This is equivalent, no? Though I suppose `/dev/stdin` is more explicit.

Reply:
It works around this: https://github.com/ruby/ruby/blob/e51435177e88fc845528dff7cf2bc2b75dd36144/ruby.c#L2333-L2335

Reply:
I've adjusted the comment to mention that